src/xopen/__init__.py

```diff
@@ -148,14 +148,9 @@ def _open_process(mode, compresslevel, threads, outfile, devnull):
         if sys.platform != 'win32':
             kwargs['close_fds'] = True
 
-        try:
```

Comment:
You’re switching from EAFP to LBYL here. That is, previously we would just try to start pigz and then gzip and then use the first one that worked, but now you’re first testing which program is available and then start it.

You may be able to convince me otherwise, but I’d prefer if the current way would be kept. It may not be so relevant in practice, but theoretically, between the time that `shutil.which` has determined which program is available and the time the program is actually started, the information on disk could have changed. Also, I think that `shutil.which` can give you false positives, that is, it could return the path to a binary that is not actually executable.

I think the way to solve this could be to add `PipedGzip...` and `PipedPigz...` classes, and then try them one after the other within `_open_gz`.

Reply:
Fine with me! :+1: